Link error on Apple mobile targets

I'm adding an SDL_Image demo to [SDL3 Sample](https://github.com/Ravbug/sdl3-sample). When building for iOS, tvOS and visionOS I get this linker error:
```
Undefined symbols for architecture arm64:
  "_CGImageSourceCreateImageAtIndex", referenced from:
      _LoadImageFromIOStream in libSDL3_image.a[3](IMG_ImageIO.o)
      _IMG_Load in libSDL3_image.a[3](IMG_ImageIO.o)
  "_CGImageSourceCreateWithDataProvider", referenced from:
      _Internal_isType in libSDL3_image.a[3](IMG_ImageIO.o)
      _LoadImageFromIOStream in libSDL3_image.a[3](IMG_ImageIO.o)
  "_CGImageSourceCreateWithURL", referenced from:
      _IMG_Load in libSDL3_image.a[3](IMG_ImageIO.o)
  "_CGImageSourceGetType", referenced from:
      _Internal_isType in libSDL3_image.a[3](IMG_ImageIO.o)
  "_UTTypeConformsTo", referenced from:
      _Internal_isType in libSDL3_image.a[3](IMG_ImageIO.o)
  "_kCGImageSourceTypeIdentifierHint", referenced from:
      _Internal_isType in libSDL3_image.a[3](IMG_ImageIO.o)
      _LoadImageFromIOStream in libSDL3_image.a[3](IMG_ImageIO.o)
  "_kUTTypeGIF", referenced from:
      _IMG_isGIF in libSDL3_image.a[3](IMG_ImageIO.o)
      _IMG_LoadGIF_IO in libSDL3_image.a[3](IMG_ImageIO.o)
  "_kUTTypeTIFF", referenced from:
      _IMG_isTIF in libSDL3_image.a[3](IMG_ImageIO.o)
      _IMG_LoadTIF_IO in libSDL3_image.a[3](IMG_ImageIO.o)
ld: symbol(s) not found for architecture arm64
```
I was able to fix it in SDL3 Sample by explicitly linking ImageIO and CoreServices:
```cmake
if (APPLE)
	find_library(IO_LIB ImageIO)
 	find_library(CS_LIB CoreServices)
	target_link_libraries(${EXECUTABLE_NAME} PUBLIC ${IO_LIB} ${CS_LIB})
endif()
```
Is it expected for users to link these libraries? Or should SDL_Image be linking them automatically?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -404,13 +404,21 @@ if(APPLE)
     if(SDLIMAGE_BACKEND_IMAGEIO)
         enable_language(OBJC)
         set(SDLIMAGE_IMAGEIO_ENABLED TRUE)
+        if(CMAKE_VERSION VERSION_LESS "3.24")
+            set(genex_FRAMEWORK "1:")
+        else()
+            set(genex_FRAMEWORK "LINK_LIBRARY:FRAMEWORK,")
+        endif()
         if(CMAKE_SYSTEM_NAME MATCHES ".*(Darwin|MacOS).*")
-            target_link_libraries(${sdl3_image_target_name} PRIVATE -Wl,-framework,ApplicationServices)
-        elseif(IOS)
+            target_link_libraries(${sdl3_image_target_name} PRIVATE $<${genex_FRAMEWORK}ApplicationServices>)
+            target_link_libraries(${sdl3_image_target_name} PRIVATE $<${genex_FRAMEWORK}Foundation>)
+        else()
             target_link_libraries(${sdl3_image_target_name} PRIVATE
-                -Wl,-framework,CoreGraphics
-                -Wl,-framework,ImageIO
-                -Wl,-framework,MobileCoreServices
+                $<${genex_FRAMEWORK}CoreGraphics>
+                $<${genex_FRAMEWORK}Foundation>
+                $<${genex_FRAMEWORK}ImageIO>
+                $<${genex_FRAMEWORK}MobileCoreServices>
+                $<${genex_FRAMEWORK}UIKit>
             )
         endif()
         target_link_libraries(${sdl3_image_target_name} PRIVATE objc)

